pkg/deployer/helm/helm.go

```diff
@@ -87,6 +90,47 @@ func New(log logr.Logger, helmconfig *helmv1alpha1.Configuration, kubeClient cli
 	}, nil
 }
 
+func createOCIClient(ctx context.Context, log logr.Logger, client client.Client, item *lsv1alpha1.DeployItem, config *helmv1alpha1.Configuration, componentsRegistryMgr *componentsregistry.Manager) (ociclient.Client, error) {
+	// resolve all pull secrets
+	secrets, err := resolveSecrets(ctx, client, item.Spec.RegistryPullSecrets)
+	if err != nil {
+		return nil, err
+	}
+
+	// always add a oci client to support unauthenticated requests
+	ociConfigFiles := make([]string, 0)
+	if config.OCI != nil {
+		ociConfigFiles = config.OCI.ConfigFiles
+	}
+	ociKeyring, err := credentials.CreateOCIRegistryKeyring(secrets, ociConfigFiles)
+	if err != nil {
+		return nil, err
+	}
+	ociClient, err := ociclient.NewClient(log,
+		utils.WithConfiguration(config.OCI),
+		ociclient.WithResolver{Resolver: ociKeyring},
+		ociclient.WithCache{Cache: componentsRegistryMgr.SharedCache()},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return ociClient, nil
+}
+
+func resolveSecrets(ctx context.Context, client client.Client, secretRefs []lsv1alpha1.ObjectReference) ([]corev1.Secret, error) {
```

Comment:
this looks like duplicated code from https://github.com/gardener/landscaper/blob/7b3f64a3298798f4e92232e6aea5be426f409575/pkg/landscaper/controllers/installations/registry.go#L78 .
I think it makes sense to add it as util function https://github.com/gardener/landscaper/blob/4edfac63889d072622f8951a401c8eee34981aab/pkg/utils/kubernetes/kubernetes.go

Reply:
Done